### DISCUSSION TITLE
Opendata quest types

### DISCUSSION THREAD

**Discussion by roptat** [streetcomplete/StreetComplete#4747] [Ideas]
Hi,

I don't think these have been discussed here before. When surveying with SC, I'd love to be able to answer quests such as "Is there a bicycle parking around here?". This type of quest could come from Open Data sources (most likely, a tool such as osmose would work great for it).

The UI would simply be a quest as usual, that would ask "Is there a… around here?". Then, two possible answers:
- "no" -> report to osmose as false positive. This ensures the question is not asked again
- "yes" -> reuse the same interface as the shop overlay to select the location. The quest automatically creates a new node at this location with the proper tags and reports it as fixed to osmose when the answer is uploaded to OSM.

To ensure the quality of created objects, there would be a quest per type of object, so SC can control which tags are added to the object. Also, we should select only object types that are very likely to be outdoors and publicly accessible. I think bicycle parking is a good example :)

Would this be useful to others?

**Comment by Cj-Malone** [comment]
I've asked similar before in #3508 and a discussion about buildings too in #2477. But I think using osmose is actually a really good idea here. It offloads a bunch of the dataset selection/license validation and conflation work to osmose, and also gives StreetComplete a single API to implement instead of one per dataset.

**Comment by Helium314** [comment]
One issue here is that it's not possible to bulk download the tags to be used ("fixeable") from Osmose. They are only available when downloading each issue individually, see https://github.com/osm-fr/osmose-frontend/issues/404

@roptat what data sources are available for this? I had a look at Mapillary map features, but it looks like there is no way to report that object was added, or found to be false positive or duplicate.

**Reply by roptat** [reply]
I was thinking about osmose because it can aggregate multiple sources, so you could add a source to SC by adding it to osmose instead. It would benefit others too. I'm not sure I understand the issue you have with osmose. Does the `fixable` tag contain more information about the object? Most data sources I've come across don't have more than a position and an id, so I was thinking that a simple request on osmose issues of a specific type (they call it class/item I think) would give us approximate positions, SC would add the tag (eg. `amenity=bicycle_parking`) from this type and correct position from user input. This would generate further quests that the user can answer since they have the object in front of them.

**Reply by Helium314** [reply]
You can find the `fixables` for some issues, e.g. [here at the bottom](https://osmose.openstreetmap.fr/en/issue/b9fd44eb-e977-14da-4de5-607481d42875) you see 2 possible tags that could be added to fix the issue (`power=pole` or `power=tower`).
These can be used to define tags for new nodes to add, which is much more versatile than just item / class combination.

@amenk had the idea of implementing a very similar thing with different scope, which currently seems stuck because a. the `fixables` are not part of Osmose bbox downloads and b. the data planned to be used is not yet availble on Osmose (though this information may be outdated)

**Reply by roptat** [reply]
Ah, the amount of tags can be quite big such as [here](https://osmose.openstreetmap.fr/en/issue/5c45175f-e98a-b71d-aad2-6eceec0aaad8). I'm not sure StreetComplete can show all that and ask the user to verify the information. Should it trust the source, or simply create a minimal object without these information if they're not visible on the ground?
